wrapanapi/systems/lenovo.py

```diff
@@ -453,3 +473,132 @@ def get_device_unique_id(device):
 
     def disconnect(self):
         self.logger.info("LenovoSystem disconnected")
+
+    def __getattr__(self, attr):
+        """
+        Tries to delegate a not found method to objects inside self._delegate
+        :raises AttributeError if ``attr`` is not found in any objects this should delegate
+        or if ``attr`` is found in 2 or more delegated classes/objects.
+        """
+        matches = []
+        for delegate_to in self._delegate:
+            if hasattr(delegate_to, attr):
+                matches.append(delegate_to)
+        if len(matches) == 1:
+            return getattr(matches[0], attr)
+        elif len(matches) > 1:
+            raise AttributeError("Ambiguous reference to attribute: {}".format(attr))
+        else:
+            raise AttributeError(attr)
+
+
+class LenovoSwitch(Entity):
+    """ Encapsulates all Lenovo Switches behavior """
+    API_PATH = "switches"
+    _switch_inventory = []
+
+    def __init__(self, switch_data, **kwargs):
+        super(LenovoSwitch, self).__init__(raw=switch_data, **kwargs)
+        self.__update_attributes()
+        LenovoSwitch._switch_inventory.append(self)
```

Comment:
This is a bad practice, and its forcing a bad pattern in `LenovoSystem.list_switches`

This is modifying the class attribute _switch_inventory, instead of modifying an instance of the class.

Why would a `LenovoSwitch` entity class maintain a list of switch objects - that list should be associated with `LenovoSystem` instance (not class!), which contains a collection of switches.

This is also forcing a pattern of staticmethods that take some identifying information as a parameter, instead of being able to instantiate a LenovoSystem object, get a list of LenovoSwitch entity objects, and then perform actions on those entity objects directly, like `delete()`.  We lose context and are forced to pass string identifiers around instead of dealing with the objects directly.